### ISSUE TITLE
BluetoothFacade: any way to add BLE Gatt Client / Server 

### ISSUE THREAD

**Issue by ThLOCP** [ainsophical/DROID_PYTHON#8]
Dear Project Members,
Developping a lot around Bluetooth Low Energy, Bluetooth Facade offcially registred in sl4a does not include the BLE Facade to access a BLE Gatt Client / Server. I have seen .java file but it is still unknown by the RPC interpreter. 
I would like to ask you two questions:
- Is there any way to to have them available in some script.jar or in .apk? I would be really eager to.
- Is there any way (detailled recipe) to rebuild sl4a? So I could maybe manage my own evolutions.
Best regards,
Thierry

**Comment by ainsophical** [comment]
Personally, I would just add a custom facade to target what you are looking for. look at the docks provided in the readme.md

**Comment by michaelrinderle** [comment]
can you provide a copy of your custom facade java file here and a segment on how it was registered with the rpc server?

**Comment by ThLOCP** [comment]
Thanks for you answer.

Since my post, I have worked hard : Thanks to kuri6536 who gave me help to start developping with the sl4a (https://github.com/kuri65536/sl4a). Now I did make work the BLE Facade. 
 
Working on the curent state of the sl4a, I became aware that it would require a huge work to bring it at the state of Today's Android API. On my side I focused only on the part I need in my use case i.e. Bluetooth low Energy and a GATT Server Client. This includes Scanning, Device connection, Service Discovery, Characteristic Reading and Writing and Notification/Indication activation. 

I will report my work to kuri65356 as soon as I have time to clean up (at least) the work I did so far. Also I guess I will it to the devzone of Nordic Semiconductor (https://devzone.nordicsemi.com/), to help developpers like me that would hardly need BLE in python that is really great, flexible, effective and quite powerful. 

Best regards
